### ISSUE TITLE
NPE for fxml-files which use the fx:root construct

### ISSUE THREAD

**Issue by mipastgt** [ciplogic/Fx2C#4]
The compiler does not seem to be able to handle fxml-files which use the fx:root construct.
An attempt results in a NullPointerException in the ReflectionResolver because it tries to look-up fx:root as a class.
For an example see this attachement:
[DataManagementView.fxml.zip](https://github.com/ciplogic/Fx2C/files/766341/DataManagementView.fxml.zip)

**Comment by ciplogic** [comment]
I will look over the next weekend. if it is an emergency, please try to figure it yourself if you can and I can assist you. 
🥇

**Comment by mipastgt** [comment]
No, it is not an emergency. I just wanted to give it try to improve the 
loading speed of our FXML-files
and stumbled over this issue because we use the fx:root construct 
throughout our whole software.
LG Michael

Am 13.02.17 um 11:52 schrieb Ciprian Khlud:

**Comment by ciplogic** [comment]
I didn't use fx:root, I will add it I think till this weekend I should be committing a patch. 

I am not sure if it works the same (I will have to test it a little), but it should mostly work as the root type defined by type attribute.

I have to look also into fx:include ...

**Comment by ciplogic** [comment]
There is a fix now. 

Take the latest master please and confirm the fix. The fx:include is not yet handled (and it may be a bit complex, as it has different namespaces that should be merged, but the non-NPE code generates now the attached java file:

[FxDataManagementView.zip](https://github.com/ciplogic/Fx2C/files/798271/FxDataManagementView.zip)

**Comment by mipastgt** [comment]
Thank you, but the attached Java file is full of bugs and does not even 
compiler.
You are generating methods which just don't exist, e.g. "setChildren" 
instead of
getChildren.add and lots of other things. I have attached a manually 
corrected
version which mostly works. The padding still seems to be wrong. I guess the
order of the named arguments is wrong. You also created an Insets 
constructor
with just two arguments which also does not exist. Also, you do not add all
items to the SplitPane. Only the first item gets added but none of the 
other two.

Michael

Am 24.02.17 um 03:50 schrieb Ciprian Khlud:

**Comment by ciplogic** [comment]
Right... there are to be expected some errors with generated Java file. But these are not necessarily fx:root issues. I will try as for this weekend to make possible to add: fx:include code and after this I will look into errors around getChildren/setChildren and so on, and other methods which fail.

Should be addressed one at a time, but they don't look related with this issue.

**Comment by mipastgt** [comment]
Am 24.02.17 um 11:15 schrieb Ciprian Khlud:
>
> Right... there are to be expected some errors with generated Java 
> file. But these are not necessarily fx:root issues.
>
Right, but they make the whole generated code unusable. I think before 
you do anything else you should make
the basic code usable. Did you ever try to actually run your generated code?

**Comment by ciplogic** [comment]
I did run the code generator and it generates sometimes good code. Not on your Fxml, of course. But with other templates I did.... I will look later into the errors as I'm at work now. 

I also have to (partially) address the fx:include to feel fine considering the issue #4 to be closeable.

If you will write instead of FxRoot the control defined by the type attribute, you should notice that the generated code should be the same, so the bug now is not related with fx:root issues, but with other issues.
